sys/base64/base64.c

```diff
@@ -56,9 +56,10 @@ static char getsymbol(unsigned char code)
     return (char)BASE64_NOT_DEFINED;
 }
 
-int base64_encode(unsigned char *data_in, size_t data_in_size, \
+int base64_encode(const void *data_in, size_t data_in_size,
                   unsigned char *base64_out, size_t *base64_out_size)
 {
+    const unsigned char *in = data_in;
```

Comment:
is this variable necesary?

Reply:
mhm, right I mean thats the point of using `void` right, to not hassle with casts.

Reply:
if I try to use the void. I get the following errors:
```
RIOT/sys/base64/base64.c: In function ‘base64_encode’:
RIOT/sys/base64/base64.c:89:28: error: pointer of type ‘void *’ used in arithmetic [-Werror=pointer-arith]
         tmpval = *(data_in + i);
                            ^
RIOT/sys/base64/base64.c:89:18: error: dereferencing ‘void’ pointer [-Werror]
         tmpval = *(data_in + i);
                  ^~~~~~~~~~~~~~
RIOT/sys/base64/base64.c:89:16: error: void value not ignored as it ought to be
         tmpval = *(data_in + i);
                ^
```

Reply:
That's right, `void*` cannot be used in pointer arithmetic.